documentation/library_guide/dynamic_selection_api/fixed_resource_policy.rst

```diff
@@ -0,0 +1,236 @@
+fixed_resource_policy
+#####################
+
+Always returns the same resource selection. ``fixed_resource_policy`` 
+is designed for two primary scenarios: (1) debugging the use of dynamic selection 
+and (2) special casing a dynamic selection capable application for a specific 
+resource is always best on a specific platform.
+
+.. code:: cpp
+
+  namespace oneapi::dpl::experimental {
+  
+    template<typename Backend=sycl_backend> 
+    class fixed_resource_policy {
+    public:
+      // useful types
+      using resource_type = typename Backend::resource_type;
+      using wait_type = typename Backend::wait_type;
+      
+      class selection_type {
+      public:
+        fixed_resource_policy<Backend> get_policy() const;
+        resource_type unwrap() const;
+      };
+      
+      // constructors
+      fixed_resource_policy(deferred_initialization_t);
+      fixed_resource_policy(::std::size_t offset=0);
+      fixed_resource_policy(const std::vector<resource_type>& u,  
+                            ::std::size_t offset=0);
+  
+      // deferred initializers
+      void initialize(::std::size_t offset=0);
+      void initialize(const std::vector<resource_type>& u, 
+                      ::std::size_t offset=0);
+                      
+      // queries
+      auto get_resources() const;
+      auto get_submission_group();
+      
+      // other implementaton defined functions...
+    };
+  
+  }
+  
+This policy can be used with all of the dynamic selection functions, such as ``select``, ``submit``,
+and ``submit_and_wait``. It also can used with ``policy_traits``.
+
+Example
+-------
+
+In the following example, a ``fixed_resource_policy`` is used when the code is
+compiled so that it selects a specific device.  When ``USE_CPU`` is defined at 
+compile-time, this example will always use the CPU queue. When ``USE_GPU`` is defined 
+at compile-time, it will always use the GPU queue. Otherwise, it uses an 
+``auto_tune_policy`` to dynamically select between these two queues. Such a scenario 
+could be used for debugging or simply to maintain the dynamic selection code even if 
+the best device to use is know for some subset of platforms.  
+
+.. code:: cpp
+
+  #include <oneapi/dpl/dynamic_selection>
+  #include <sycl/sycl.hpp>
+  #include <iostream>
+
+  namespace ex = oneapi::dpl::experimental;
+
+  int main() {
+    std::vector<sycl::queue> r { sycl::queue{sycl::cpu_selector_v},
+                                 sycl::queue{sycl::gpu_selector_v} };
+
+    const size_t N = 10000;
+    std::vector<float> av(N, 0.0);
+    std::vector<float> bv(N, 0.0);
+    std::vector<float> cv(N, 0.0);
+    for (int i = 0; i < N; ++i) {
+      av[i] = bv[i] = i;
+    }
+
+  #if USE_CPU
+    ex::fixed_resource_policy p{r};    // (1)
+  #elif USE_GPU
+    ex::fixed_resource_policy p{r, 1}; // (2)

```

Comment:
I am not sure if it's clear enough that `1` means GPU (on the user platform).

Should we clarify it somehow? Probably adding extra comment?

Reply:
Added to the comment. Hopefully this addressed your concern.